vtex-sites/nextjs.store
# feat: CMS Error Boundaries

## What's the purpose of this pull request?
This PR adds error boundaries around CMS sections so bugs in individual sections do not affect the whole page

## How does it work?
A new error boundary called SectionBoundary was added. This Boundary is rendered by `RenderPageSections`, wrapping each CMS section. In case of error:
```
if production: renders null
else: renders children
```

This make it so when developing, developers will receive the error, but in production, that section will disappear.

> Note that error boundaries do not work in SSR, so errors on SSR will still prevent the page from being rendered and a 500 will be thrown.

## How to test it?
Add an error to a section, say shelf, and run this with yarn develop and yarn build. On develop nothing should have changed, but on production the section should just not render
